app/models/manageiq/providers/redhat/infra_manager/inventory/strategies/v4.rb

```diff
@@ -20,6 +20,19 @@ def host_targeted_refresh(target)
       end
     end
 
+    def template_targeted_refresh(target)
+      ems.with_provider_connection(VERSION_HASH) do |connection|
+        @connection = connection
+        template_id = get_uuid_from_target(target)
+        res = {}
+        res[:cluster] = collect_clusters
```

Comment:
we share 3 resources between vms and templates. What do you think about:

    def vm_targeted_refresh(target)
      ems.with_provider_connection(VERSION_HASH) do |connection|
        @connection = connection
        vm_id = get_uuid_from_target(target)
        res = {}
        res[:cluster] = collect_clusters
        res[:datacenter] = collect_datacenters
        res[:storage] = target.storages.empty? ? collect_storages : collect_storage(target.storages.map { |s| get_uuid_from_target(s) })
        if target.template ? target_template(res, vm_id) : target_vm(res, vm_id)
        res
      end
    end

    def target_template(res, template_id)
        res[:template] = collect_template_by_uuid(template_id)
    end

    def target_vm(res, template_id)
        res[:template] = search_templates("vm.id=#{vm_id}")
        res[:vm] = collect_vm_by_uuid(vm_id)
    end

Reply:
While this code is shorter, I do not think its clearer. You call vm_target_refresh but deal with a template.

Reply:
We deal with a template in this code anyway.

Reply:
sure, but then we call it: "template_targeted_refresh"

Reply:
Not really, please look [here](https://github.com/ManageIQ/manageiq-providers-ovirt/blob/master/app/models/manageiq/providers/redhat/infra_manager/inventory/strategies/v4.rb#L32)

Reply:
@pkliczewski you'll have to play a similar trick for https://github.com/ManageIQ/manageiq-providers-ovirt/pull/165/files#diff-ec4f927ab832832041a9d1574a4fa6a2R21 which will make it in that case less readable

Reply:
Correct, it is yet another copy/paste where we change 3 lines when copying 15.